spec/controllers/steps/case/charges_controller_spec.rb

```diff
@@ -4,6 +4,63 @@
   let(:form_class) { Steps::Case::ChargesForm }
   let(:decision_tree_class) { Decisions::CaseDecisionTree }
 
+  let(:crime_application) { CrimeApplication.create }
+  let(:existing_case) { Case.create(crime_application: crime_application)}
+  let(:charge_record) { Charge.create(case: existing_case)}
+
+  context 'adding a new offence date' do
+    let(:form_class_params_name) { form_class.name.underscore }
+
+    it 'has the expected step name' do
+      expect(
+        subject
+      ).to receive(:update_and_advance).with(
+        form_class, 
+        record: charge_record, 
+        as: :add_offence_date
+      )
+
+      put :update, params: { 
+        id: crime_application.id, 
```

Comment:
It has no real impact, but no need to call `#id` explicitly, as just passing the record will implicitly call `#to_param` on the record, which will return (usually) the ID (in this case UUID).
This is valid anywhere in the rails app where we build URLs, and also in named urls/paths.